pkg/genericapiserver/config.go

```diff
@@ -180,6 +200,29 @@ func NewConfig(options *options.ServerRunOptions) *Config {
 		}
 	}
 
+	var secureServingInfo *ServingInfo
+	if options.SecurePort > 0 {
+		secureServingInfo = &ServingInfo{
+			BindAddress: net.JoinHostPort(options.BindAddress.String(), strconv.Itoa(options.SecurePort)),
+			ServerCert: CertInfo{
+				CertFile: options.TLSCertFile,
+				KeyFile:  options.TLSPrivateKeyFile,
+			},
+			ClientCA: options.ClientCAFile,
+		}
+		if options.TLSCertFile == "" && options.TLSPrivateKeyFile == "" {
```

Comment:
this is the point at which we know whether we should generate a self-signed one or not